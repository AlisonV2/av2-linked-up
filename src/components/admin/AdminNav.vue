<template>
  <header class="header admin-navbar">
    <div class="container-fluid">
      <nav class="navbar-container">
        <ul class="navbar-menu" :class="{ active: isActive }">
          <li class="nav-link">
            <a class="logout-link"
              ><span class="d-none d-sm-inline mx-3" @click.prevent="logout"
                >Logout</span
              >
              <i class=" fs-3 bi bi-box-arrow-right"></i
            ></a>
          </li>
        </ul>
      </nav>
    </div>
  </header>
</template>

<script>
/**
 * @exports AdminNavbar
 * @type {Component}
 * @vue-data {boolean} isActive
 * @vue-event logout
 */
export default {
  name: 'Navbar',
  data() {
    return {
      isActive: false,
    };
  },
  methods: {
    /**
     * @description Dispatch logout store action
     * Redirect to Home page if the current route requires authentication
     * @method logout
     */
    logout() {
      this.$store.dispatch('logout');
      if (this.$route.meta.requiresAuth) {
        this.$router.push({ name: 'Home' });
      }
    },
  },
};
</script>
<style lang="scss" scoped>
.admin-navbar {
  background-color: $dark;
  height: 3.5rem;
  .navbar-container {
    display: flex;
    justify-content: flex-end;
  }
  .nav-item {
    color: $light;
  }
  span {
    color: $light;
  }
  .logout-link {
    display: flex;
    align-items: center;
  }
}
</style>
