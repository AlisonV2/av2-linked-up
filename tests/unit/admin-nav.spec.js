import { shallowMount } from '@vue/test-utils';
import AdminNav from '@/components/admin/AdminNav.vue';
import { createStore } from 'vuex';

const actions = {
  logout: jest.fn(),
};

const store = createStore({
  actions,
});

const $router = {
  push: jest.fn(),
};

const config = {
  data() {
    return {
      isActive: false,
    };
  },
  global: {
    plugins: [store],
    mocks: {
      $route: {
        meta: {
          requiresAuth: true,
        },
      },
      $router,
    },
  },
};

/**
 * @module AdminNavTest
 */
describe('AdminNav.vue', () => {
  it('AdminNav Snapshot', () => {
    const component = shallowMount(AdminNav, config);
    expect(component.element).toMatchSnapshot();
  });

  it('Check requiresAuth check redirection', async () => {
    const component = shallowMount(AdminNav, config);
    component.find('.test-logout').trigger('click');
    await component.vm.$nextTick();
    expect($router.push).toHaveBeenCalled();
  });

  it('Check requiresAuth check redirection', async () => {
    const component = shallowMount(AdminNav, config);

    component.find('.test-logout').trigger('click');
    await component.vm.$nextTick();
    expect($router.push).toHaveBeenCalledWith({ name: 'Home' });
  });

  it('Dispatch an action on click', async () => {
    const component = shallowMount(AdminNav, {
      data() {
        return {
          isActive: false,
        };
      },
      global: {
        plugins: [store],
        mocks: {
          $route: {
            meta: {
              requiresAuth: false,
            },
          },
          $router,
        },
      },
    });

    component.find('.test-logout').trigger('click');
    await component.vm.$nextTick();
    expect(actions.logout).toHaveBeenCalled();
  });
});
