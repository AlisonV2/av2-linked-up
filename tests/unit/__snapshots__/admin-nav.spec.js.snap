// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminNav.vue AdminNav Snapshot 1`] = `
<header
  class="header admin-navbar"
>
  <div
    class="container-fluid"
  >
    <nav
      class="navbar-container"
    >
      <ul
        class="navbar-menu"
      >
        <li
          class="nav-link"
        >
          <a
            class="logout-link"
          >
            <span
              class="d-none d-sm-inline mx-3 test-logout"
            >
              Logout
            </span>
            <i
              class="fs-3 bi bi-box-arrow-right"
            />
          </a>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;
